### PR TITLE
Simple mode for datasets, update dataset headers

### PR DESCRIPTION
**Pull request by kulyk**
Updates question header component for datasets and questions based on datasets:

* when a dataset is open, the header is more similar to "Simple Mode" when viewing a table 
* when a dataset is open, its collection is displayed instead of a database name
* when a question based on a dataset is open, the dataset name and its collection are shown instead of a database and tables

### To Verify

1. Open a dataset (you can create one following the steps from #18702)
2. Make the page header has a collection link instead of a database link
2. Add a filter / summarize something
3. Make sure the URL is changed from `/question/:id-:slug` to `/question#hash` (moved to ad-hoc mode)
4. Click the "Save" button
5. Make sure you're not offered to replace the dataset, only to save a new question
6. Save the question
7. Make sure the shows the dataset name and its collection instead of database and tables

### Demo

https://user-images.githubusercontent.com/17258145/141836214-2ea7f8e8-112a-4d07-9aa7-feb90ec0a80d.mov

### CODE REVIEW

**Comment by kulyk** [comment]
@metabase-bot run visual tests

**Review by mazameli — APPROVED** [review]
🎉

File: frontend/src/metabase/query_builder/components/view/QuestionDataSource.jsx [hunk 1]
```diff
@@ -5,26 +5,112 @@ import {
...
-      variant={subHead ? "subhead" : "head"}
       {...props}
+      parts={[
+        <HeadBreadcrumbs.Badge
+          key="dataset-collection"
+          to={Urls.collection(collection)}
+          icon="dataset"
+          inactiveColor="text-light"
+        >
+          {collection?.name || "Our analytics"}
```
Comment (daltojohnso) [inline frontend/src/metabase/query_builder/components/view/QuestionDataSource.jsx:104]: need a translation here for `Our analytics`

File: frontend/src/metabase/query_builder/components/view/ViewHeader.jsx [hunk 2]
```diff
@@ -295,10 +309,93 @@ function AhHocQuestionLeftSide(props) {
...
+
+DatasetCollectionBadge.propTypes = {
+  dataset: PropTypes.object.isRequired,
+};
+
+function DatasetCollectionBadge({ dataset }) {
+  const { collection } = dataset.card();
+  return (
+    <HeadBreadcrumbs.Badge to={Urls.collection(collection)} icon="dataset">
+      {collection?.name || "Our analytics"}
```
Comment (daltojohnso) [inline frontend/src/metabase/query_builder/components/view/ViewHeader.jsx:389]: need a translation here

**Review by daltojohnso — CHANGES_REQUESTED** [review]
nice work. just a few minor things.

also, I noticed if you're starting from an id url (`/question/123`) and you make a change so that the URL updates, there are situations where the saved button won't show up:

https://user-images.githubusercontent.com/13057258/141873903-76adbe25-c536-4267-9e0c-b0cca0dd5fc2.mov

not sure if you want to handle that in this PR but just pointing it out

**Comment by kulyk** [comment]
@daltojohnso kudos for giving it a proper test. Thanks to the issue you found I realized the implementation was pretty much unstable. Could you please give it one more look? The main changes are mostly about QB actions file

**Comment by mazameli** [comment]
@anton, when I try to open the sidebar for one of my preexisting datasets, I'm seeing a stacktrace error pop up:

<details>
<summary>Full stacktrace</summary>
in RevisionBatchedDescription (created by QuestionActivityTimeline)
    in div (created by Timeline)
    in div (created by Timelinestyled__ItemBody)
    in Timelinestyled__ItemBody (created by Timeline)
    in div (created by Timelinestyled__TimelineItem)
    in Timelinestyled__TimelineItem (created by Timeline)
    in div (created by Timelinestyled__TimelineContainer)
    in Timelinestyled__TimelineContainer (created by Timeline)
    in Timeline (created by QuestionActivityTimeline)
    in div (created by DrawerSectionstyled__Children)
    in DrawerSectionstyled__Children (created by ControlledDrawerSection)
    in div (created by DrawerSectionstyled__Transformer)
    in DrawerSectionstyled__Transformer (created by ControlledDrawerSection)
    in div (created by DrawerSectionstyled__Container)
    in DrawerSectionstyled__Container (created by ControlledDrawerSection)
    in ControlledDrawerSection (created by DrawerSection)
    in DrawerSection (created by QuestionActivityTimeline)
    in QuestionActivityTimeline (created by Connect(QuestionActivityTimeline))
    in Connect(QuestionActivityTimeline) (created by LoadingAndErrorWrapper)
    in LoadingAndErrorWrapper (created by EntityListLoader)
    in EntityListLoader (created by Connect(EntityListLoader))
    in Connect(EntityListLoader) (created by _class2)
    in _class2 (created by EntityType)
    in EntityType (created by Connect(EntityType))
    in Connect(EntityType) (created by Connect(QuestionActivityTimeline))
    in Connect(QuestionActivityTimeline) (created by EntityListLoader)
    in EntityListLoader (created by Connect(EntityListLoader))
    in Connect(EntityListLoader) (created by _class2)
    in _class2 (created by EntityType)
    in EntityType (created by Connect(EntityType))
    in Connect(EntityType) (created by Connect(QuestionActivityTimeline))
    in Connect(QuestionActivityTimeline) (created by QuestionDetailsSidebarPanel)
    in div (created by QuestionDetailsSidebarPanelstyled__Container)
    in QuestionDetailsSidebarPanelstyled__Container (created by QuestionDetailsSidebarPanel)
    in QuestionDetailsSidebarPanel (created by QuestionDetailsSidebar)
    in QuestionDetailsSidebar (created by View)
    in div (created by ViewSidebarstyled__ViewSidebarContent)
    in ViewSidebarstyled__ViewSidebarContent (created by ViewSidebar)
    in aside (created by ViewSidebarstyled__ViewSidebarAside)
    in ViewSidebarstyled__ViewSidebarAside (created by ViewSidebar)
    in ViewSidebar (created by View)
    in div (created by View)
    in div (created by View)
    in div (created by View)
    in View (created by ExplicitSize[View])
    in ExplicitSize[View] (created by QueryBuilder)
    in QueryBuilder (created by FitViewport)
    in FitViewport
    in Unknown (created by Title[])
    in Title[] (created by Title[Title[]])
    in Title[Title[]] (created by Connect(Title[Title[]]))
    in Connect(Title[Title[]]) (created by RouterContext)
    in Unknown (created by UserIsAuthenticated(Component))
    in UserIsAuthenticated(Component) (created by Connect(UserIsAuthenticated(Component)))
    in Connect(UserIsAuthenticated(Component)) (created by MetabaseIsSetup(Connect(UserIsAuthenticated(Component))))
    in MetabaseIsSetup(Connect(UserIsAuthenticated(Component))) (created by Connect(MetabaseIsSetup(Connect(UserIsAuthenticated(Component)))))
    in Connect(MetabaseIsSetup(Connect(UserIsAuthenticated(Component)))) (created by RouterContext)
    in div (created by App)
    in ScrollToTop (created by withRouter(ScrollToTop))
    in withRouter(ScrollToTop) (created by App)
    in App (created by Connect(App))
    in Connect(App) (created by Title[Connect(App)])
    in Title[Connect(App)] (created by RouterContext)
    in RouterContext (created by Router)
    in Router
    in ThemeProvider
    in DragDropContextProvider
    in Provider
</details>

image

**Review by daltojohnso — APPROVED** [review]
Looks good! The bug Maz noted above looks to be in the Timeline/Revision description code, so idk if related, so I'll go ahead and give this the 👍

**Comment by kulyk** [comment]
> @kulyk, when I try to open the sidebar for one of my preexisting datasets, I'm seeing a stacktrace error pop up. It appears to be happening on all my GUI-based datasets, but not my SQL ones.

> The bug Maz noted above looks to be in the Timeline/Revision description code, so idk if related

@mazameli @daltojohnso this is most likely caused by the changes in this PR. We're modifying GUI dataset's card, so this can cause some regressions in the QB

Maz, I couldn't easily reproduce the issue but found a different one related to revisions. Could you please give it another look? Or maybe you remember the steps that led to the error?

**Comment by mazameli** [comment]
> Maz, I couldn't easily reproduce the issue but found a different one related to revisions. Could you please give it another look? Or maybe you remember the steps that led to the error?

Yeah, I'll definitely give it another look. The repro step was just opening the sidebar, but I wonder if it had to do with some preexisting reivision history item.

**Review by mazameli — APPROVED** [review]
So I can only reproduce that bug with datasets that were built using a previous version of this branch. Any new dataset I create does not display an error when I open the sidebar. It makes me slightly concerned that previously-created datasets are reliably demonstrating this error for me, still, but probably not a true issue.

**Pull request merged**